tests/capi/unittest_capi_inference.cc

```diff
@@ -1437,6 +1438,175 @@ TEST (nnstreamer_capi_src, callback_invalid_param_02_n)
   EXPECT_EQ (status, ML_ERROR_NONE);
 }
 
+/**
+ * @brief Check decoded orange.png with raw data.
+ */
+static void
+check_orange_output (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data)
+{
+  int status, fd;
+  size_t raw_data_size, data_size;
+  ssize_t data_read;
+  void *raw_data;
+  gchar *orange_raw_file;
+
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  orange_raw_file = g_build_filename (
+      root_path, "tests", "test_models", "data", "orange.raw", NULL);
+  ASSERT_TRUE (g_file_test (orange_raw_file, G_FILE_TEST_EXISTS));
+
+  fd = open (orange_raw_file, G_FILE_TEST_EXISTS);
+  EXPECT_TRUE (fd >= 0);
+
+  raw_data_size = lseek (fd, 0, SEEK_END);
+  raw_data = g_malloc0 (raw_data_size);
+  ASSERT_TRUE (raw_data != NULL);
+
+  status = lseek (fd, 0, SEEK_SET);
+  EXPECT_EQ (status, 0);
+
+  status = ml_tensors_data_get_tensor_data (data, 0, (void **) &data, &data_size);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  data_read = read (fd, raw_data, data_size);
+  EXPECT_EQ ((size_t) data_read, data_size);
+
+  status = 0;
+  for (size_t i = 0; i < data_size; ++i) {
+    if (*(((uint8_t *) data) + i) != *(((uint8_t *) raw_data) + i)) {
+      status = 1;
+      break;
+    }
+  }
+
+  EXPECT_EQ (status, 0);
+
+  g_free (raw_data);
+  g_free (orange_raw_file);
+  close (fd);
+}
+
+/**
+ * @brief Test NNStreamer pipeline src (appsrc with png file)
+ */
+TEST (nnstreamer_capi_src, pngfile)
+{
+  int status, fd;
+  size_t png_file_size, data_size;
+  ssize_t data_read;
+  void *data = NULL;
+
+  ml_pipeline_h handle;
+  ml_pipeline_sink_h sinkhandle;
+  ml_pipeline_src_h srchandle;
+  ml_pipeline_state_e state;
+
+  ml_tensors_info_h in_info;
+  ml_tensor_dimension in_dim;
+  ml_tensors_data_h input = NULL;
+
+  gchar *orange_png_file, *pipeline;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  /* start pipeline test with valid model file */
+  orange_png_file = g_build_filename (
+      root_path, "tests", "test_models", "data", "orange.png", NULL);
+  ASSERT_TRUE (g_file_test (orange_png_file, G_FILE_TEST_EXISTS));
+
+  fd = open (orange_png_file, O_RDONLY);
+  EXPECT_TRUE (fd >= 0);
+
+  pipeline = g_strdup_printf (
+    "appsrc name=srcx caps=image/png ! pngdec ! videoconvert ! videoscale ! video/x-raw,format=RGB,width=224,height=224,framerate=0/1 ! tensor_converter ! tensor_sink name=sinkx sync=false async=false"
+  );
+
+  /* construct pipeline */
+  status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  EXPECT_EQ (status ,ML_ERROR_NONE);
```

Comment:
Minor style change:
```suggestion
  EXPECT_EQ (status, ML_ERROR_NONE);
```
+ L1536

Reply:
Thanks! I fixed the code.